peripherals/tim/tim12_common_wl.yaml

```diff
@@ -0,0 +1,200 @@
+      TI2: [2, "CCx channel is configured as input, ICx is mapped on TI2"]
+      TRC: [3, "CCx channel is configured as input, ICx is mapped on TRC"]
+
+  CCMR[12]_Output:
+    OC?PE:
+      Disabled: [0, "Preload register on CCRx disabled. New values written to CCRx are taken into account immediately"]
+      Enabled: [1, "Preload register on CCRx enabled. Preload value is loaded into active register on each update event"]
+    OC?CE:
+      Disabled: [0, "OCxRef is not affected by the ocref_clr_int signal"]
+      Enabled: [1, "OCxRef is cleared as soon as a High level is detected on ocref_clr_int signal"]
+    OC?FE:
+      Disabled: [0, "Fast output disabled"]
+      Enabled: [1, "Fast output enabled"]
+    CC?S:
+      "Output": [0, "CCx channel is configured as output"]
+      TI1: [1, "CCx channel is configured as input, ICx is mapped on TI1"]
+      TI2: [2, "CCx channel is configured as input, ICx is mapped on TI2"]
+      TRC: [3, "CCx channel is configured as input, ICx is mapped on TRC"]
+    OC?M:
+      Frozen: [0, "The comparison between the output compare register TIMx_CCRy and the counter TIMx_CNT has no effect on the outputs. Alt (OCxM[3]): Channel active until trigger is detected on TRGI signal"]
+      ActiveOnMatch: [1, "Set channel to active level on match. OCyREF signal is forced high when the counter matches the capture/compare register. Alt (OCxM[3]): OC1REFC outputs OC1REF when the counter is counting up, OC2REF when it is counting down"]
+      InactiveOnMatch: [2, "Set channel to inactive level on match. OCyREF signal is forced low when the counter matches the capture/compare register"]
+      Toggle: [3, "OCyREF toggles when TIMx_CNT=TIMx_CCRy"]
+      ForceInactive: [4, "OCyREF is forced low. Alt (OCxM[3]): OC1REFC is the logical OR between OC1REF and OC2REF"]
+      ForceActive: [5, "OCyREF is forced high. Alt (OCxM[3]): OC1REFC outputs OC1REF when the counter is counting up, OC2REF when it is counting down"]
+      PwmMode1: [6, "In upcounting, channel is active as long as TIMx_CNT<TIMx_CCRy else inactive. In downcounting, channel is inactive as long as TIMx_CNT>TIMx_CCRy else active. Alt (OCxM[3]): OC1REFC outputs OC1REF when the counter is counting up, OC2REF when it is counting down"]
+      PwmMode2: [7, "Inversely to PwmMode1. Alt (OCxM[3]): Assymetric PWM Mode 2: OC1REFC outputs OC1REF when the counter is counting up, OC2REF when it is counting down"]
+    OC?M_3:
+      NormalModes: [0, "Choose low modes (0-7)"]
+      AlternateModes: [1, "Choose high modes (8-15)"]
+
+  CCER:
+    "CC?NP":
+      Negative: [0, "Negative polarity"]
+      Positive: [1, "Positive polarity"]
+    "CC?P":
+      RisingEdge: [0, "Noninverted/rising edge"]
+      FallingEdge: [1, "Inverted/falling edge"]
+    "CC?E":
+      Disabled: [0, "Capture disabled"]
+      Enabled: [1, "Capture enabled"]
+
+  PSC:
+    PSC: [0, 0xFFFF]
+
+  DCR:
+    DBL: [0, 0b10001]
```

Comment:
Safe `bits()` methods are only generated when all possible bit values are permissible, but in this case you're specifying that only values 0-17 are allowed, and since 18-31 could fit in the register but aren't allowed, svd2rust does not generate a safe function for it.

It's arguable whether that's a legitimate thing to make unsafe, but it's how svd2rust works at the moment. You could instead put an enum of all eighteen permitted values and then it would generate safe methods for each of them, but it's a bit boring...

Reply:
Ok thanks, for the moment will leave like this then.